test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs

```diff
@@ -38,23 +74,132 @@ public void DthStartup_GetProjectInformation(string flavor, string os, string ar
+            var projectName = "EmptyConsoleApp";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            var testProject = _fixture.GetTestProjectPath(projectName);
+
+            using (var server = DthTestServer.Create(runtimeHomePath, testProject))
+            using (var client = new DthTestClient(server, 0))
+            {
+                client.SetProtocolVersion(0);
+
+                Assert.Throws<TimeoutException>(() =>
+                {
+                    client.DrainTillFirst(ProtocolManager.NegotiationMessageTypeName);
+                });
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DthCompilation_GetDiagnostics_OnEmptyConsoleApp(string flavor, string os, string architecture)
+        {
+            var projectName = "EmptyConsoleApp";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            var testProject = _fixture.GetTestProjectPath(projectName);
+
+            using (var server = DthTestServer.Create(runtimeHomePath, testProject))
+            using (var client = new DthTestClient(server, 0))
+            {
+                // Drain the inital messages
+                client.Initialize(testProject);
+                client.SendPayLoad("GetDiagnostics", new { });
+
+                var message = client.DrainTillFirst("AllDiagnostics", TimeSpan.FromSeconds(10));
+                message.EnsureSource(server, client);
+
+                var payload = message.Payload as JArray;
+                Assert.Equal(3, payload.OfType<JObject>().Count());
+
+                foreach (var df in payload.OfType<JObject>())
+                {
+                    var errors = (JArray)df["Errors"];
+                    Assert.False(errors.Any(), $"Unexpected compilation errors {string.Join(", ", errors.Select(e => e.Value<string>()))}");
+
+                    var warnings = (JArray)df["Warnings"];
+                    Assert.False(warnings.Any(), $"Unexpected compilation warnings {string.Join(", ", warnings.Select(e => e.Value<string>()))}");
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DthCompilation_GetCompiledAssembly_OnEmptyConsoleApp(string flavor, string os, string architecture)
+        {
+            var projectName = "EmptyLibrary";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+
+            string testProject;
+            using (_fixture.CreateDisposableTestProject(projectName, runtimeHomePath, out testProject))
+            using (var server = DthTestServer.Create(runtimeHomePath, testProject))
+            using (var client = new DthTestClient(server, 0))
+            {
+                // Drain the inital messages
+                client.Initialize(testProject);
+
+                var before = client.DrainTillFirst("Dependencies");
```

Comment:
for which framework? This isn't very nice. I think we should be asserting specific things. The order of the framework messages shouldnt' matter